src/DayPicker.tsx

```diff
@@ -425,42 +437,52 @@ export function DayPicker(initialProps: DayPickerProps) {
                       className={classNames[UI.Dropdowns]}
                       style={styles?.[UI.Dropdowns]}
                     >
-                      {captionLayout === "dropdown" ||
-                      captionLayout === "dropdown-months" ? (
-                        <components.MonthsDropdown
-                          className={classNames[UI.MonthsDropdown]}
-                          aria-label={labelMonthDropdown()}
-                          classNames={classNames}
-                          components={components}
-                          disabled={Boolean(props.disableNavigation)}
-                          onChange={handleMonthChange(calendarMonth.date)}
-                          options={dropdownMonths}
-                          style={styles?.[UI.Dropdown]}
-                          value={dateLib.getMonth(calendarMonth.date)}
-                        />
-                      ) : (
-                        <span>
-                          {formatMonthDropdown(calendarMonth.date, dateLib)}
-                        </span>
-                      )}
-                      {captionLayout === "dropdown" ||
-                      captionLayout === "dropdown-years" ? (
-                        <components.YearsDropdown
-                          className={classNames[UI.YearsDropdown]}
-                          aria-label={labelYearDropdown(dateLib.options)}
-                          classNames={classNames}
-                          components={components}
-                          disabled={Boolean(props.disableNavigation)}
-                          onChange={handleYearChange(calendarMonth.date)}
-                          options={dropdownYears}
-                          style={styles?.[UI.Dropdown]}
-                          value={dateLib.getYear(calendarMonth.date)}
-                        />
-                      ) : (
-                        <span>
-                          {formatYearDropdown(calendarMonth.date, dateLib)}
-                        </span>
-                      )}
+                      {dropdownTypes.map((type) => {
```

Comment:
This new feature will also adapt the month and year labels, not just the `dropdowns`.
Could you rename `dropdownTypes` to a more generic name that refers to the date-time format of the `locale`?